gaudi/gaudi_spi_fw_upgrade_job.yaml

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: gaudi-spi-firmware-upgrade
+  namespace: gaudi-validation
+spec:
+  template:
+    metadata:
+    spec:
+      restartPolicy: Never
+      serviceAccountName: gaudi-fw-upgrade-sa
+      containers:
+        - name: gaudi-spi-firmware-upgrade
+          securityContext:
+            privileged: true
+          image: image-registry.openshift-image-registry.svc:5000/gaudi-validation/gaudi-spi-fw-upgrade:1.18.0-524
+          command: [ "hl-fw-loader", "-y" ]
+          imagePullPolicy: IfNotPresent
```

Comment:
We should set 
imagePullPolicy: Always
and make sure always using the latest FW.

Reply:
How about eROM FW handling? Should we also need to upgrade the eROM FW?
How about CPLD FW?

Reply:
@hershpa  please review the PR. :-)

Reply:
We also need to figure out how to handle the OAM power cycle to make SPI FW upgrading work and whether we can avoid rebooting the node.

Reply:
> How about eROM FW handling? Should we also need to upgrade the eROM FW?

eROM FW was frozen in 1.12 release and is not relevant anymore. 

> We also need to figure out how to handle the OAM power cycle to make SPI FW upgrading work and whether we can avoid rebooting the node.

Upgrading the SPI FW does not need node reboot. It flashes the card with latest FW without node reboots.

Reply:
I think we need to make sure the steps flashing the FWs on the stand along server are well documented firstly.